### DISCUSSION TITLE
Language Specification: Abstract static methods

### DISCUSSION THREAD

**Discussion by dkearns** [vim/vim#13462] [General]
I think these are a pretty conceptually confused feature in the context of the language as it stands.  Rather than expend energy debating the details at this time, and if no one is particularly attached to the idea, could we consider just dropping them?

Adding them at a later time presents no problems.

**Comment by errael** [comment]
> consider just dropping them?

I don't see what role they play. Can someone describe a use case? Seems it would be better to drop unused/unneeded features.

Seems like an abstract class or interface is at the root of a polymorphic model. Following is typical abstract class usage:
```
vim9script

abstract class A
    abstract def ObjFoo()
endclass

class C extends A
    def ObjFoo()
        echo "C's ObjFoo"
    enddef
endclass

var o = C.new()

# <<< Cannot access static methods defined in A
def F(oarg: A)
    oarg.ObjFoo()
enddef
F(o)
```
The function `def F(oarg: A)` is the key; it shows how an abstract class can be used. If the `abstract class A` has `abstract static def ClassFoo()` what role does it play? How is it used?

**Comment by dkearns** [comment]
Closed by https://github.com/vim/vim/pull/13466

Thanks @yegappan
